index.js

```diff
@@ -1,22 +1,39 @@
 const waterfall = require('run-waterfall')
+const pullAsyncMap = require('pull-stream/throughs/async-map')
 
 module.exports = cliPrompter
 
 const promptTypes = require('./types')
 
 function cliPrompter (questions, cb) {
+  if (arguments.length === 0) {
+    return pullAyncScan({}, (values, question, cb) => {
+      const prompter = getPrompter(question)
+      prompter({ question, values }, cb)
+    })
+  }
+
   waterfall([
     cb => cb(null, {}), // start the waterfall with values
-    ...questions.map(promptQuestion)
+    ...questions.map(question => {
+      const prompter = getPrompter(question)
+      return (values, cb) => {
+        prompter({ question, values }, cb)
+      }
+    })
   ], cb)
 }
 
-function promptQuestion (question) {
-  const { type } = question
-  const Prompter = promptTypes[type]
-
-  return (values, cb) => {
-    Prompter({ question, values }, cb)
-  }
+function getPrompter ({ type }) {
+  return promptTypes[type]
 }
 
+function pullAyncScan (value, accumulator) {
+  return pullAsyncMap((next, cb) => {
+    accumulator(value, next, (err, nextValue) => {
+      if (err) return cb(err)
+      value = nextValue
+      cb(null, value[Object.keys(value)[0]])
```

Comment:
hypothetically, what if a prompt changed more than one key?

this is related to my question:

> i'm wondering how values accumulate in this module, like should future prompts know about all previous answers, and should they be able to even modify previous answers?

at the moment, a "prompt" is a function of shape

```js
function ({ question, values }, cb) {
  cb(err, nextValues)
}
```

i did it so a prompt could

- know about all previous answers
- be able to modify previous answers
- be able to add more than one key

but maybe this is a terrible idea.

what do you think?